src/components/Icon/IconStyles.tsx

```diff
@@ -3,30 +3,37 @@ import styled from 'styled-components'
 
 import { Spacing } from '../../types/Spacing'
 import { marginCss } from '../common/Spacing/SpacingStyles'
+import { isSpacing } from '../common/Spacing/utils'
 
 interface StyledIconProps {
-	size: Spacing | string | number
+	$size: Spacing | string | number
 }
 
-export const StyledIcon = styled(SVG)<StyledIconProps>`
-	display: inline-block;
-	${({ size, theme }): string => {
-		if (Spacing.includes(size as unknown as Spacing)) {
+export const StyledIconContainer = styled.span<StyledIconProps>`
+	display: inline-flex;
+
+	${({ $size, theme }): string => {
+		if (isSpacing($size)) {
 			return `
-				width: ${theme.$pc.spacing[size as unknown as Spacing]}px;
-				height: ${theme.$pc.spacing[size as unknown as Spacing]}px;
+				width: ${theme.$pc.spacing[$size as unknown as Spacing]}px;
```

Comment:
Do we still need the casts if we have the type guard?

Reply:
Of course we don't :-) I fixed it.